File: C/island_perimeter.c
int countshareSides(int** grid, int row, int col, int gridSize, int* gridColSize) {
    int shareSides = 0;
        
    if(row-1 > - 1) { 
        if(grid[row-1][col] == 1)
            shareSides ++;
    }
    if(row+1 < gridSize) { 
        if(grid[row+1][col] == 1)
            shareSides ++;
    }
    if(col-1 > - 1) { 
        if(grid[row][col-1] == 1)
            shareSides ++;
    }
    if(col+1 < *gridColSize) { 
        if(grid[row][col+1] == 1)
            shareSides ++;
    }
        
    return shareSides;
}

int islandPerimeter(int** grid, int gridSize, int* gridColSize){
    int perimeter = 0;
        
    for(int i = 0; i < gridSize; i++) {
        for(int j = 0; j < *gridColSize; j++) {
            if(grid[i][j] == 1) 
                perimeter += (4 - countshareSides(grid, i, j, gridSize, gridColSize));
        }
    }
        
    return perimeter;
}

File: Python3/island_perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        perimeter = 0
        
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                if grid[i][j] == 1:
                    perimeter += (4 - self.countshareSides(grid, i, j))
        
        return perimeter        
    
    def countshareSides(self, grid, row, col) -> int:
        shareSides = 0
        
        for num_row, num_col in [(row + 1, col), (row - 1, col), (row, col + 1), (row, col - 1)]:
            if num_row > - 1 and num_row < len(grid) and num_col > - 1 and num_col < len(grid[0]):
                if grid[num_row][num_col] == 1: shareSides += 1
        
        return shareSides
